build-system/tasks/helpers.js

```diff
@@ -818,27 +817,29 @@ async function getDependencies(entryPoint, options) {
 }
 
 /**
- * @param {*} sourcemapsFile
+ * @param {string} sourcemapsFile
+ * @param {*} minifiedMap
  * @param {*} options
- * @return {*}
+ * @return {string}
  */
-function massageSourcemaps(sourcemapsFile, options) {
-  const sourcemaps = JSON.parse(sourcemapsFile);
-  sourcemaps.sources = sourcemaps.sources.map((source) => {
+function massageSourcemaps(sourcemapsFile, minifiedMap, options) {
+  const remapped = remapping(
+    [minifiedMap, sourcemapsFile],
+    () => null,
+    !argv.full_sourcemaps
```

Comment:
nit: we can move the `argv.full_sourcemaps` check up to to the esbuild call by adding in the build flag: `sourcesContent: !!argv.full_sourcemaps`.

Reply:
In both now.